packages/slice-machine/components/Simulator/index.tsx

```diff
@@ -58,22 +67,50 @@ export default function Simulator() {
     return <div />;
   }
 
-  const sliceView = useMemo(
-    () => [
-      {
-        sliceID: component.model.id,
-        variationID: variation.id,
+  const sharedSlice = useMemo(
+    () => Slices.fromSM(component.model),
+    [component.model]
+  );
+
+  const initialContent = useMemo<SharedSliceContent>(
+    () =>
+      component.mock?.[0] ||
+      (defaultSharedSliceContent(variation.id) as SharedSliceContent),
+    [component.mock, variation.id]
+  );
+
+  const [editorContent, setContent] = useState(initialContent);
+  const initialApiContent = useMemo(
+    () =>
+      renderSliceMock(sharedSlice, editorContent) as {
+        id: string;
+        [k: string]: unknown;
       },
-    ],
-    [component.model.id, variation.id]
+    []
+  );
+  const apiContent = useThrottle(
```

Comment:
the throttle seems really weird, you seem to set the value at every throttle interval, meaning rendering the editor content, but that'll happen all the time here, even if editor content doesn't change. That would trigger a lot of conversion for no reason if there are no user actions. It should happen only if editor changes but with a throttle on it to prevent too many transformations in a given window of time.